### PR TITLE
fix(Core/Spells): Fixed Wolverine Bite. Source: TrinityCore.

### LINKED ISSUE #10293: Chimaera as hunter pet never uses "Wolverine Bite"
Original CC Issue: https://github.com/chromiecraft/chromiecraft/issues/2841

### Current Behaviour

When selected in the pet talent tree the pet can/should use the ability "Wolverine Bite" after a critical hit. This never happens. No matter how many times it crits (on dummys or mobs in the world) it never goes on cooldown and when looking at Skada logs it never shows up there either, furthermore when I try to use it manually after I see a crit it gives me the text "The ability requires combo points".

### Expected Blizzlike Behaviour

The in-game description of the ability is "A fierce attack causing 305 damage, modified by pet level, that your pet can use after it makes a critical attack. Cannot be dodged, blocked or parried."

### Source

From Wowpedia: "[Wolverine Bite]: This talent is now enabled when the pet lands a critical strike rather than from the target dodging the pet's attacks. In addition, this talent no longer has a prerequisite."

https://wowpedia.fandom.com/wiki/Patch_3.3.0

### Steps to reproduce the problem

All I do is send the pet to attack, with the ability enabled and let it attack.

### Extra Notes

https://wowgaming.altervista.org/aowow/?spell=53508#modified-by

### AC rev. hash/commit

chromiecraft/azerothcore-wotlk@259b913

### Operating system

Windows 10 x64 Pro

### Custom changes or Modules

- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-chat-transmitter](https://github.com/azerothcore/mod-chat-transmitter)
- [mod-chromie-xp](https://github.com/azerothcore/mod-chromie-xp)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-eluna-lua-engine](https://github.com/azerothcore/mod-eluna-lua-engine)
- [mod-ip-tracker](https://github.com/azerothcore/mod-ip-tracker)
- [mod-low-level-arena](https://github.com/azerothcore/mod-low-level-arena)
- [mod-multi-client-check](https://github.com/azerothcore/mod-multi-client-check)
- [mod-pvp-titles](https://github.com/azerothcore/mod-pvp-titles)
- [mod-pvpstats-announcer](https://github.com/azerothcore/mod-pvpstats-announcer)
- [mod-queue-list-cache](https://github.com/azerothcore/mod-queue-list-cache)
- [mod-server-auto-shutdown](https://github.com/azerothcore/mod-server-auto-shutdown)
- [mod-save-inventory](https://github.com/Nyeriah/mod-save-inventory)
- [lua-carbon-copy](https://github.com/55Honey/Acore_CarbonCopy)
- [lua-exchange-npc](https://github.com/55Honey/Acore_ExchangeNpc)
- [lua-custom-worldboss](https://github.com/55Honey/Acore_CustomWorldboss)
- [lua-level-up-reward](https://github.com/55Honey/Acore_LevelUpReward)
- [lua-recruit-a-friend](https://github.com/55Honey/Acore_RecruitAFriend)
- [lua-send-and-bind](https://github.com/55Honey/Acore_SendAndBind)
- [lua-temp-announcements](https://github.com/55Honey/Acore_TempAnnouncements)
- [lua-zonecheck](https://github.com/55Honey/acore_Zonecheck)
- [lua-zone-debuff](https://github.com/55Honey/Acore_ZoneDebuff)

### PR DESCRIPTION
**Pull request by UltraNix**
Fixed #10293

## Issues Addressed:

- Closes #10293
- Closes https://github.com/chromiecraft/chromiecraft/issues/2841

## Tests Performed:

- Not tested

## How to Test the Changes:

More info [here](https://github.com/azerothcore/azerothcore-wotlk/issues/10293)

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by acidmanifesto** [comment]
TC Cherry pick for reference: https://github.com/TrinityCore/TrinityCore/commit/489478b74d96c2ea940a2316fd11f8e81991cf07
Author: Treeston <treeston.mmoc@gmail.com>

File: src/server/game/Entities/Unit/Unit.cpp [hunk 1]
```diff
@@ -16357,6 +16355,10 @@ void Unit::UpdateReactives(uint32 p_time)
                     if (getClass() == CLASS_WARRIOR && GetTypeId() == TYPEID_PLAYER)
                         ClearComboPoints();
                     break;
+                case REACTIVE_WOLVERINE_BITE:
+                    if (IsHunterPet())
+                        ClearComboPoints();
```
Comment (Nefertumm) [inline src/server/game/Entities/Unit/Unit.cpp:16360]: brackets

**Review by Nefertumm — APPROVED** [review]

**Comment by Nefertumm** [comment]
Tested and working as intended. Now the pet casts Wolverine Bite.
image
image

**Comment by FrancescoBorzi** [comment]
funny Codefactor throwing an error completely unrelated to these changes..

**Pull request merged**